db/repair.cc

```diff
@@ -410,10 +410,11 @@ class Repairer {
           std::unique_ptr<InternalIterator>(mem->NewRangeTombstoneIterator(ro)),
           &meta, cfd->internal_comparator(),
           cfd->int_tbl_prop_collector_factories(), cfd->GetID(), cfd->GetName(),
-          {}, kMaxSequenceNumber, kNoCompression, CompressionOptions(), false,
-          nullptr /* internal_stats */, TableFileCreationReason::kRecovery,
-          nullptr /* event_logger */, 0 /* job_id */, Env::IO_HIGH,
-          nullptr /* table_properties */, -1 /* level */, current_time);
+          {}, kMaxSequenceNumber, nullptr /*snapshot_checker*/, kNoCompression,
```

Comment:
why does not repair need snapshot checker? does not it flush memtables when they are full?

Reply:
Transaction is not involved in repair (let me know if I'm wrong) and snapshot_checker is not needed.

Reply:
Could repair result into memtable flush? Does it involve any memtable insert?

Reply:
I was assuming these will be no preparing transaction on db open, but I was wrong since there can be preparing transactions recovering from logs. Let me think..

Reply:
Looks like the repair code doesn't support transactions, since it simply replay logs and write them into sst file but don't even look at prepare/commit markers.